### PR TITLE
[onert] Fix bug that doesn't round indices values in acl kernels

### PR DESCRIPTION
**Pull request by ragmani**
From https://github.com/Samsung/ONE/pull/4331#discussion_r493974899

This commit fixes bug that doesn't round indices values in acl kernels
  - Fix bug
  - Add some nnfw api tests for OneHot

### CODE REVIEW

File: compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp [hunk 1]
```diff
@@ -90,6 +90,15 @@ Status validate_arguments(const ITensorInfo *indices, const ITensorInfo *depth,
 
   return Status{};
 }
+
+template <typename U, typename Enable = void> bool isOnValue(U) { return true; }
+
+template <typename U, typename std::enable_if<std::is_signed<U>::value>::type>
+bool isOnValue(U value)
+{
+  ARM_COMPUTE_ERROR_ON(value < -1);
```
Comment (glistening) [inline compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp:99]: @ragmani It seems to throw error on `value < -1`. I think it should return `false`. 
Also, I think `indice >= depth` is valid input that we should not set to on_value.

Comment (ragmani) [reply compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp:99]: I misunderstood. I will fix it. Thanks for your pointing out.

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-debug

**Comment by ragmani** [comment]
@nnfw-bot test copyright-checker

**Comment by hseok-oh** [comment]
It needs rebase to pass copyright-checker

**Comment by ragmani** [comment]
> It needs rebase to pass copyright-checker

Thanks for your information. I did it.

**Review by hseok-oh — APPROVED** [review]

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**